### PR TITLE
[1LP][RFR] Add miq polarion, template for polarion_tools.local.yaml

### PR DESCRIPTION
**Pull request by mshriver**
Add miq polarion, template for polarion_tools.local.yaml

Update requirements for ruamel.yaml and polarion

Make casecomponent required for polarion docblocks

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit d406a8fecff531a393ce635932337872caa12490:

`cfme/scripting/polarion.py`:
- :red_circle: Line 1:1: **F401** *'argparse' imported but unused*
- :red_circle: Line 2:1: **F401** *'sys' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by jawatts** [comment]
PR #8509 needs to be merged before this PR. I currently have to kick off jenkins build with 1 provider and 1 appliance to get an accurate pass rate.

File: cfme/scripting/polarion.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+@click.option('--polarion-project-id',
+              help='The polarion project ID to inject',
+              default=None)
+@click.option('--polarion-url',
+              help='The polarion URL to inject',
+              default=None)
+@click.option('--template-format',
+              help='The format for the test template, y-stream CFME digits is formatted in',
+              default='ImportTestRunTemplate')
+def populate(
```
Comment (izapolsk) [inline cfme/scripting/polarion.py]: do you like such formatting ?

Comment (mshriver) [reply cfme/scripting/polarion.py]: Kinda, not so much, I dropped them to a newline. Its quite verbose in general trying to decorate `click` methods with these.

Feel free to format as you like, and I'll merge the PR :)

**Pull request merged**